File: jacspy/python/jacs/testing.py
"""
JACS Test Utilities

Provides a pytest fixture for creating ephemeral JacsClient instances
that are automatically cleaned up after each test.

Usage:
    # In your test file or conftest.py:
    from jacs.testing import jacs_agent

    def test_sign_and_verify(jacs_agent):
        signed = jacs_agent.sign_message({"test": True})
        result = jacs_agent.verify(signed.raw_json)
        assert result.valid
"""

import pytest

from .client import JacsClient


@pytest.fixture
def jacs_agent():
    """Pytest fixture that yields an ephemeral JacsClient.

    The client is created in-memory with no disk I/O or environment
    variables required. It is reset after the test completes.
    """
    client = JacsClient.ephemeral()
    yield client
    client.reset()

File: archive/native/jacsnpm/testing.ts
/**
 * JACS Test Helpers
 *
 * Utilities for creating ephemeral clients in tests. Each client gets its own
 * in-memory agent -- no config files, no key files, no env vars needed.
 *
 * @example
 * ```typescript
 * import { createTestClient, createTestClientSync } from '@hai.ai/jacs/testing';
 *
 * // Async (preferred)
 * const client = await createTestClient('pq2025');
 * const signed = await client.signMessage({ hello: 'test' });
 * const result = await client.verify(signed.raw);
 * assert(result.valid);
 *
 * // Sync
 * const client2 = createTestClientSync('pq2025');
 * const signed2 = client2.signMessageSync({ hello: 'test' });
 * const result2 = client2.verifySync(signed2.raw);
 * assert(result2.valid);
 * ```
 */

import { JacsClient } from './client';

export { JacsClient };

/**
 * Create an ephemeral JacsClient for testing (async). No files or env vars needed.
 *
 * @param algorithm - Signing algorithm (default: "pq2025")
 * @returns A fully-initialized JacsClient backed by an in-memory agent
 */
export async function createTestClient(algorithm?: string): Promise<JacsClient> {
  return JacsClient.ephemeral(algorithm ?? 'pq2025');
}

/**
 * Create an ephemeral JacsClient for testing (sync). No files or env vars needed.
 *
 * @param algorithm - Signing algorithm (default: "pq2025")
 * @returns A fully-initialized JacsClient backed by an in-memory agent
 */
export function createTestClientSync(algorithm?: string): JacsClient {
  return JacsClient.ephemeralSync(algorithm ?? 'pq2025');
}
